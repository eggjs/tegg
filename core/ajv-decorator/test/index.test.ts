import { expect, it } from 'vitest';
import * as types from '../src/index.ts';

it('should export stable', async () => {
  expect(types).toMatchSnapshot();
});
