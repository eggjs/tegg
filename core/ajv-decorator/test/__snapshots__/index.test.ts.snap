// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should export stable 1`] = `
{
  "AjvInvalidParamError": [Function],
  "Any": [Function],
  "Array": [Function],
  "AsyncIterator": [Function],
  "Awaited": [Function],
  "BigInt": [Function],
  "Boolean": [Function],
  "Capitalize": [Function],
  "Clone": [Function],
  "CloneRest": [Function],
  "CloneType": [Function],
  "Composite": [Function],
  "Const": [Function],
  "Constructor": [Function],
  "ConstructorParameters": [Function],
  "Date": [Function],
  "Deref": [Function],
  "Enum": [Function],
  "Exclude": [Function],
  "ExcludeFromMappedResult": [Function],
  "ExcludeFromTemplateLiteral": [Function],
  "Extends": [Function],
  "ExtendsCheck": [Function],
  "ExtendsFromMappedKey": [Function],
  "ExtendsFromMappedResult": [Function],
  "ExtendsResolverError": [Function],
  "ExtendsResult": {
    "0": "Union",
    "1": "True",
    "2": "False",
    "False": 2,
    "True": 1,
    "Union": 0,
  },
  "ExtendsUndefinedCheck": [Function],
  "Extract": [Function],
  "ExtractFromMappedResult": [Function],
  "ExtractFromTemplateLiteral": [Function],
  "FormatRegistry": {
    "Clear": [Function],
    "Delete": [Function],
    "Entries": [Function],
    "Get": [Function],
    "Has": [Function],
    "Set": [Function],
  },
  "Function": [Function],
  "Hint": Symbol(TypeBox.Hint),
  "Increment": [Function],
  "Index": [Function],
  "IndexFromMappedKey": [Function],
  "IndexFromMappedResult": [Function],
  "IndexFromPropertyKey": [Function],
  "IndexFromPropertyKeys": [Function],
  "IndexPropertyKeys": [Function],
  "InstanceType": [Function],
  "Integer": [Function],
  "Intersect": [Function],
  "IntersectEvaluated": [Function],
  "Intrinsic": [Function],
  "IntrinsicFromMappedKey": [Function],
  "IsTemplateLiteralExpressionFinite": [Function],
  "IsTemplateLiteralFinite": [Function],
  "Iterator": [Function],
  "JavaScriptTypeBuilder": [Function],
  "JsonTypeBuilder": [Function],
  "KeyOf": [Function],
  "KeyOfFromMappedResult": [Function],
  "KeyOfPattern": [Function],
  "KeyOfPropertyEntries": [Function],
  "KeyOfPropertyKeys": [Function],
  "KeyOfPropertyKeysToRest": [Function],
  "Kind": Symbol(TypeBox.Kind),
  "KindGuard": {
    "IsAny": [Function],
    "IsArray": [Function],
    "IsAsyncIterator": [Function],
    "IsBigInt": [Function],
    "IsBoolean": [Function],
    "IsConstructor": [Function],
    "IsDate": [Function],
    "IsFunction": [Function],
    "IsInteger": [Function],
    "IsIntersect": [Function],
    "IsIterator": [Function],
    "IsKind": [Function],
    "IsKindOf": [Function],
    "IsLiteral": [Function],
    "IsLiteralBoolean": [Function],
    "IsLiteralNumber": [Function],
    "IsLiteralString": [Function],
    "IsMappedKey": [Function],
    "IsMappedResult": [Function],
    "IsNever": [Function],
    "IsNot": [Function],
    "IsNull": [Function],
    "IsNumber": [Function],
    "IsObject": [Function],
    "IsOptional": [Function],
    "IsPromise": [Function],
    "IsProperties": [Function],
    "IsReadonly": [Function],
    "IsRecord": [Function],
    "IsRecursive": [Function],
    "IsRef": [Function],
    "IsRegExp": [Function],
    "IsSchema": [Function],
    "IsString": [Function],
    "IsSymbol": [Function],
    "IsTemplateLiteral": [Function],
    "IsThis": [Function],
    "IsTransform": [Function],
    "IsTuple": [Function],
    "IsUint8Array": [Function],
    "IsUndefined": [Function],
    "IsUnion": [Function],
    "IsUnknown": [Function],
    "IsUnsafe": [Function],
    "IsVoid": [Function],
  },
  "Literal": [Function],
  "Lowercase": [Function],
  "Mapped": [Function],
  "MappedFunctionReturnType": [Function],
  "MappedKey": [Function],
  "MappedResult": [Function],
  "Never": [Function],
  "Not": [Function],
  "Null": [Function],
  "Number": [Function],
  "Object": [Function],
  "Omit": [Function],
  "OmitFromMappedKey": [Function],
  "OmitFromMappedResult": [Function],
  "Optional": [Function],
  "OptionalFromMappedResult": [Function],
  "OptionalKind": Symbol(TypeBox.Optional),
  "Parameters": [Function],
  "Partial": [Function],
  "PartialFromMappedResult": [Function],
  "PatternBoolean": "(true|false)",
  "PatternBooleanExact": "^(true|false)$",
  "PatternNever": "(?!.*)",
  "PatternNeverExact": "^(?!.*)$",
  "PatternNumber": "(0|[1-9][0-9]*)",
  "PatternNumberExact": "^(0|[1-9][0-9]*)$",
  "PatternString": "(.*)",
  "PatternStringExact": "^(.*)$",
  "Pick": [Function],
  "PickFromMappedKey": [Function],
  "PickFromMappedResult": [Function],
  "Promise": [Function],
  "Readonly": [Function],
  "ReadonlyFromMappedResult": [Function],
  "ReadonlyKind": Symbol(TypeBox.Readonly),
  "ReadonlyOptional": [Function],
  "Record": [Function],
  "Recursive": [Function],
  "Ref": [Function],
  "RegExp": [Function],
  "Required": [Function],
  "RequiredFromMappedResult": [Function],
  "Rest": [Function],
  "ReturnType": [Function],
  "SetComplement": [Function],
  "SetDistinct": [Function],
  "SetIncludes": [Function],
  "SetIntersect": [Function],
  "SetIntersectMany": [Function],
  "SetIsSubset": [Function],
  "SetUnion": [Function],
  "SetUnionMany": [Function],
  "Strict": [Function],
  "String": [Function],
  "Symbol": [Function],
  "TemplateLiteral": [Function],
  "TemplateLiteralExpressionGenerate": [Function],
  "TemplateLiteralFiniteError": [Function],
  "TemplateLiteralGenerate": [Function],
  "TemplateLiteralGenerateError": [Function],
  "TemplateLiteralParse": [Function],
  "TemplateLiteralParseExact": [Function],
  "TemplateLiteralParserError": [Function],
  "TemplateLiteralPattern": [Function],
  "TemplateLiteralPatternError": [Function],
  "TemplateLiteralSyntax": [Function],
  "TemplateLiteralToUnion": [Function],
  "Transform": [Function],
  "TransformDecodeBuilder": [Function],
  "TransformEncodeBuilder": [Function],
  "TransformEnum": {
    "toEnumCase": "toEnumCase",
    "toLowerCase": "toLowerCase",
    "toUpperCase": "toUpperCase",
    "trim": "trim",
    "trimEnd": "trimEnd",
    "trimLeft": "trimLeft",
    "trimRight": "trimRight",
    "trimStart": "trimStart",
  },
  "TransformKind": Symbol(TypeBox.Transform),
  "Tuple": [Function],
  "Type": {
    "Any": [Function],
    "Array": [Function],
    "AsyncIterator": [Function],
    "Awaited": [Function],
    "BigInt": [Function],
    "Boolean": [Function],
    "Capitalize": [Function],
    "Composite": [Function],
    "Const": [Function],
    "Constructor": [Function],
    "ConstructorParameters": [Function],
    "Date": [Function],
    "Deref": [Function],
    "Enum": [Function],
    "Exclude": [Function],
    "Extends": [Function],
    "Extract": [Function],
    "Function": [Function],
    "Index": [Function],
    "InstanceType": [Function],
    "Integer": [Function],
    "Intersect": [Function],
    "Iterator": [Function],
    "KeyOf": [Function],
    "Literal": [Function],
    "Lowercase": [Function],
    "Mapped": [Function],
    "Never": [Function],
    "Not": [Function],
    "Null": [Function],
    "Number": [Function],
    "Object": [Function],
    "Omit": [Function],
    "Optional": [Function],
    "Parameters": [Function],
    "Partial": [Function],
    "Pick": [Function],
    "Promise": [Function],
    "Readonly": [Function],
    "ReadonlyOptional": [Function],
    "Record": [Function],
    "Recursive": [Function],
    "Ref": [Function],
    "RegExp": [Function],
    "Required": [Function],
    "Rest": [Function],
    "ReturnType": [Function],
    "Strict": [Function],
    "String": [Function],
    "Symbol": [Function],
    "TemplateLiteral": [Function],
    "Transform": [Function],
    "Tuple": [Function],
    "Uint8Array": [Function],
    "Uncapitalize": [Function],
    "Undefined": [Function],
    "Union": [Function],
    "Unknown": [Function],
    "Unsafe": [Function],
    "Uppercase": [Function],
    "Void": [Function],
  },
  "TypeBoxError": [Function],
  "TypeGuard": {
    "IsAny": [Function],
    "IsArray": [Function],
    "IsAsyncIterator": [Function],
    "IsBigInt": [Function],
    "IsBoolean": [Function],
    "IsConstructor": [Function],
    "IsDate": [Function],
    "IsFunction": [Function],
    "IsInteger": [Function],
    "IsIntersect": [Function],
    "IsIterator": [Function],
    "IsKind": [Function],
    "IsKindOf": [Function],
    "IsLiteral": [Function],
    "IsLiteralBoolean": [Function],
    "IsLiteralNumber": [Function],
    "IsLiteralString": [Function],
    "IsLiteralValue": [Function],
    "IsMappedKey": [Function],
    "IsMappedResult": [Function],
    "IsNever": [Function],
    "IsNot": [Function],
    "IsNull": [Function],
    "IsNumber": [Function],
    "IsObject": [Function],
    "IsOptional": [Function],
    "IsPromise": [Function],
    "IsProperties": [Function],
    "IsReadonly": [Function],
    "IsRecord": [Function],
    "IsRecursive": [Function],
    "IsRef": [Function],
    "IsRegExp": [Function],
    "IsSchema": [Function],
    "IsString": [Function],
    "IsSymbol": [Function],
    "IsTemplateLiteral": [Function],
    "IsThis": [Function],
    "IsTransform": [Function],
    "IsTuple": [Function],
    "IsUint8Array": [Function],
    "IsUndefined": [Function],
    "IsUnion": [Function],
    "IsUnionLiteral": [Function],
    "IsUnknown": [Function],
    "IsUnsafe": [Function],
    "IsVoid": [Function],
    "TypeGuardUnknownTypeError": [Function],
  },
  "TypeRegistry": {
    "Clear": [Function],
    "Delete": [Function],
    "Entries": [Function],
    "Get": [Function],
    "Has": [Function],
    "Set": [Function],
  },
  "Uint8Array": [Function],
  "Uncapitalize": [Function],
  "Undefined": [Function],
  "Union": [Function],
  "UnionEvaluated": [Function],
  "Unknown": [Function],
  "Unsafe": [Function],
  "Uppercase": [Function],
  "ValueGuard": {
    "IsArray": [Function],
    "IsAsyncIterator": [Function],
    "IsBigInt": [Function],
    "IsBoolean": [Function],
    "IsDate": [Function],
    "IsFunction": [Function],
    "IsIterator": [Function],
    "IsNull": [Function],
    "IsNumber": [Function],
    "IsObject": [Function],
    "IsRegExp": [Function],
    "IsString": [Function],
    "IsSymbol": [Function],
    "IsUint8Array": [Function],
    "IsUndefined": [Function],
  },
  "Void": [Function],
}
`;
